docs/features/enumerations.rst

```diff
@@ -0,0 +1,217 @@
+Enumerations
+============
+
+Description
+-----------
+
+DSG can handle the generation of enumerations in your `.proto` files in three ways:
+
+- Using a Serializer or Model field with TextChoices (limited features)
+- Using an Annotated Serializer or Model field (all features)
+- Using FieldDict in the `grpc_action` decorator (all features)
+
+Unless you are using enums in a FieldDict directly, you should be using `TextChoices` or `IntegerChoices` as defined in the Django documentation `here <https://docs.djangoproject.com/en/5.1/ref/models/fields/#enumeration-types>`_.
+
+Using a FieldDict
+-----------------
+
+In a FieldDict, you can specify an Enum for the type. This will generate the corresponding enum in the `.proto` file.
+
+.. code-block:: python
+
+    @grpc_action(
+        request=[{"name": "enum_example", "type": MyEnum}],
+        response=[{"name": "value", "type": "string"}],
+    )
+
+Here is the generated enum in `.proto` file:
+
+.. code-block:: protobuf
+
+    message Request {
+        MyEnum.Enum enum_example = 1;
+    }
+
+    message MyEnum {
+        enum Enum {
+            ENUM_UNSPECIFIED = 0;
+            FIRST = 1;
+            SECOND = 2;
+        }
+    }
+
+Using a Serializer or Model field with TextChoices
```

Comment:
this should be the first option displayed in this doc as it is the commended one